### ISSUE TITLE
Inconsistent results when running prettier cli and prettier-vscode

### ISSUE THREAD

**Issue by ben-x9** [prettier/prettier-vscode#671]
Format on save:
```
"nyc": {
    "include": ["src/**/*.ts", "src/**/*.tsx"],
    "extension": [".ts", ".tsx"],
    "exclude": ["storybook-static"],
    "reporter": ["json", "html", "text"],
    "sourceMap": true,
    "all": true
  },
```

`prettier --write \"**/*.{ts*(x),js*(on),scss}\"` :
```
"nyc": {
    "include": [
      "src/**/*.ts",
      "src/**/*.tsx"
    ],
    "extension": [
      ".ts",
      ".tsx"
    ],
    "exclude": [
      "storybook-static"
    ],
    "reporter": [
      "json",
      "html",
      "text"
    ],
    "sourceMap": true,
    "all": true
  },
```

When formatting a JSON file (package.json)

prettier 1.15.2
prettier-vscode 1.7.2

.prettierrc.js
```
module.exports = {
  bracketSpacing: false,
  semi: false
}
```

**Comment by centigrade-thomas-becker** [comment]
Related: https://github.com/prettier/prettier-vscode/issues/641

**Comment by scorphus** [comment]
Any updates!? This renders the plugin useless.

**Comment by CiGit** [comment]
There is a special rule for package.json but it should not be different from prettier's CLI 
`JSON.stringify(package.json, null, 2)`

**Comment by robphoenix** [comment]
I also have inconsistencies when formatting my react code.

Prettier formats as so:

```js
React.useEffect(
    () => {
```

and the extension as so:
```js
React.useEffect(() => {
```

**Comment by ntotten** [comment]
We now resolve configuration and file type using prettiers built in tools. This issue should be resolved as of version 3.0 (preview). If you can install the preview and see if you still have issues. If you do feel free to reopen this issue. https://marketplace.visualstudio.com/items?itemName=Prettier.prettier-vscode
